### ISSUE TITLE
Scroller for Bootstrap 4 with long text in columns results in malformed horizontal scrolling

### ISSUE THREAD

**Issue by mboratko** [DataTables/DataTables#1037]
**Demonstration:** http://jsfiddle.net/xpvt214o/39190/

**Browser:** Chrome 65.0.3325.181

**Description:** Turning on the scroller for a table with cells which have a long text string, which normally would be wrapped, ends up turning off wrapping. Even if this much was intentional, there is an additional bug - scrolling to the right does not scroll the header as well, resulting in misaligned columns.

**Solution:* My solution was to override the `white-space: nowrap;` line from `scroller.bootstrap4.min.css` to `white-space: normal;`. I'm not sure if this has unintended consequences (please let me know if it does) however it worked for my use case.

**Comment by mboratko** [comment]
Actually, the point of this bug is that I did *not* want horizontal scrolling.

As shown in the jsfiddle, just adding the scroller resulted in horizontal scrolling which didn't work properly.

**Comment by DataTables** [comment]
Scroller fundamentally requires that the rows all be the same height, which is why it includes white-space: nowrap.

**Comment by mboratko** [comment]
I see. (Although, for my use case, turning off the wrapping worked just fine, at least for now, even though the rows were no longer all the same height.)

In that case it is a bug because the form of scrolling that gets turned on doesn't work correctly - if you scroll to the right the header does not scroll with it.

Two proposed solutions would be:

1. Leave wrapping turned on, but then adjust the other rows to be the same as the max height of the wrapped rows.
2. Leave wrapping turned off, but if it notices that this causes a horizontal scroll bar to appear then it should automatically turn `scrollX: true` so that the buggy behavior doesn't occur.

Of the two I prefer the first. That being said, I seem to have no issue in the production use where I've just set `white-space: normal;` even though the rows are not the same size.

**Comment by mboratko** [comment]
Also - **thanks for this great library!**

I've only just found it, but I love that I was able to just drop it in and have it take over the HTML tables on my site.

**Comment by mboratko** [comment]
As you mentioned, I did start getting bad formatting as a result of changin `white-space: nowrap` to `white-space: normal`, so I changed it back.

I would consider this a bug, however, since there is no condition under which the exhibited result in the jsfiddle is what someone would want.

**Comment by DataTables** [comment]
Agreed - although I'm not sure how much DataTables can do to stop this out of the box. Doing an auto height max would be horribly slow since you need to measure the height of every row and then set it for every row.

There is the [`scroller.rowHeight` option](https://datatables.net/reference/option/scroller.rowHeight) which you can use to set it manually though.
